core/Piranha/Services/MediaService.cs

```diff
@@ -47,28 +48,36 @@ public MediaService(IMediaRepository repo, IParamService paramService, IStorage
             _cache = cache;
         }
 
+        //Separated this into its own thing in case it needed to get reused elsewhere.
+        private async Task<IEnumerable<Media>> _getFast(IEnumerable<Guid> ids)
+        {
+            var guids = ids as Guid[] ?? ids.ToArray();
+            var partial = (_cache != null ? guids.Select(c => _cache.Get<Media>(c.ToString())) : Enumerable.Empty<Media>()).Where(c => c != null).ToArray();
+            var missingIds = guids.Except(partial.Select(c => c.Id));
+            var returns = partial.Concat((await _repo.GetAllByIdAsync(missingIds)).Select(c =>
+            {
+                OnLoad(c);
+                return c;
+            })).ToArray();
+            return returns;
+        }
```

Comment:
The only this that should be added here is that the input params coming from `_repo.GetAll` is sorted. If I don't read the code wrong the returned collection from this method will not be in the same order as the input but rather (sorted partial result from cache) + (sorted partial result from repository).

Reply:
Yes, I forgot to add the sorting call there by mistake. I'll add it after the partial.Concat call.